### PR TITLE
Fix audit warnings for (some) linuxbrew-only formulae

### PR DESCRIPTION
**Pull request by maxim-belkin**
Fix several (not all) warnings produced by `brew audit --strict` for formulae that have "linuxbrew" tag.

### CODE REVIEW

File: Formula/libfuse.rb [hunk 1]
```diff
@@ -35,6 +35,6 @@ def install
         return fuse_main(argc, argv, &ops, NULL);
       }
     EOS
-    system "$CC $CPPFLAGS -DFUSE_USE_VERSION=26 -D_FILE_OFFSET_BITS=64 fuse-test.c $LDFLAGS -lfuse"
+    system "$CC", "$CPPFLAGS", "-DFUSE_USE_VERSION=26", "-D_FILE_OFFSET_BITS=64", "fuse-test.c", "$LDFLAGS", "-lfuse"
```
Comment (sjackman) [inline Formula/libfuse.rb]: This version of `system` won't execute a shell, and so the environment variables won't be expanded. You need…
```ruby
system ENV["CC"], ENV["CPPFLAGS"], "-DFUSE_USE_VERSION=26", …
```

File: Formula/libuuid.rb [hunk 2]
```diff
@@ -1,4 +1,5 @@
 class Libuuid < Formula
+  desc "Portable uuid C library"
```
Comment (sjackman) [inline Formula/libuuid.rb]: Please capitalize UUID.

**Comment by sjackman** [comment]
Two small changes. Thanks, Maxim!

File: Formula/libfuse.rb (hunk 1 above)
Comment (maxim-belkin) [reply Formula/libfuse.rb]: `ENV.cc`?

File: Formula/libuuid.rb (hunk 2 above)
Comment (maxim-belkin) [reply Formula/libuuid.rb]: ok

File: Formula/libfuse.rb (hunk 1 above)
Comment (sjackman) [reply Formula/libfuse.rb]: In fact remove `$CPPFLAGS` and `$LDFLAGS`. They shouldn't be necessary.

Comment (sjackman) [reply Formula/libfuse.rb]: You may need to add
```ruby
`-I`, include, `-L`, lib
```

File: Formula/libfuse.rb [hunk 3]
```diff
@@ -35,6 +35,6 @@ def install
         return fuse_main(argc, argv, &ops, NULL);
       }
     EOS
-    system "$CC $CPPFLAGS -DFUSE_USE_VERSION=26 -D_FILE_OFFSET_BITS=64 fuse-test.c $LDFLAGS -lfuse"
+    system ENV.cc, ENV["$CPPFLAGS"], "-DFUSE_USE_VERSION=26", "-D_FILE_OFFSET_BITS=64", "fuse-test.c", "$LDFLAGS", "-lfuse"
```
Comment (sjackman) [inline Formula/libfuse.rb]: This test failed. https://circleci.com/gh/Linuxbrew/homebrew-core/19458
Try …
```ruby
system ENV.cc, "-DFUSE_USE_VERSION=26", "-D_FILE_OFFSET_BITS=64", "fuse-test.c", "-lfuse"
```

**Comment by maxim-belkin** [comment]
`elfutils` failed with: `/bin/bash: m4: command not found`... adding depends_on "m4"

File: Formula/libtirpc.rb [hunk 4]
```diff
@@ -1,16 +1,16 @@
...
+  sha256 "86c3a78fc1bddefa96111dd233124c703b22a78884203c55c3e06b3be6a0fd5e"
   # tag "linuxbrew"
 
   bottle do
     cellar :any_skip_relocation
     sha256 "7c093d0948f7794df11cae0a0f0e974e8b0734d46beb2f7977db39d84aa16720" => :x86_64_linux # glibc 2.19
   end
 
-  depends_on "krb5" => :optional unless OS.mac?
+  depends_on "krb5"
```
Comment (maxim-belkin) [inline Formula/libtirpc.rb]: not sure why it was `optional` before.

**Comment by maxim-belkin** [comment]
some tests failed for `elfutils`
```
FAIL: run-strip-strmerge.sh
FAIL: run-readelf-self.sh
FAIL: run-backtrace-native-core.sh
FAIL: run-disasm-x86.sh
FAIL: run-strip-reloc.sh
```

**Comment by maxim-belkin** [comment]
Reason: no `-g` flag during compilation. See https://bugzilla.redhat.com/show_bug.cgi?id=811878#c6

**Comment by sjackman** [comment]
> Reason: no -g flag during compilation. See https://bugzilla.redhat.com/show_bug.cgi?id=811878#c6

Good sleuthing. It's not easy to undisable `-g` because it's removed by `ENV.refurbish_args` with no way that I know of to turn it off.
https://github.com/Linuxbrew/brew/blob/4b3c66a7bfc423fbcdc6c73495761a2a949c82fa/Library/Homebrew/shims/super/cc#L171

I can only suggest disabling `make check`.

**Comment by sjackman** [comment]
There's some more `brew audit` warnings on CircleCI.
```
brew audit libgpm
  * The URL http://www.nico.schottelius.org/software/gpm/ should use HTTPS rather than HTTP
  * Stable: The URL http://www.nico.schottelius.org/software/gpm/archives/gpm-1.20.7.tar.bz2 should use HTTPS rather than HTTP
brew audit libtirpc
  * The URL http://sourceforge.net/projects/libtirpc/ should use HTTPS rather than HTTP
```
https://circleci.com/gh/Linuxbrew/homebrew-core/19502

File: Formula/libtirpc.rb (hunk 4 above)
Comment (sjackman) [reply Formula/libtirpc.rb]: Does it fail to build without `krb5`? Can this dependency be removed?

**Comment by maxim-belkin** [comment]
What about disabling tests that fail?

**Comment by sjackman** [comment]
Sure if there's an easy way to do that.

**Comment by maxim-belkin** [comment]
in 839d521 I added a few more tests that fail in my linuxbrew container

**Comment by maxim-belkin** [comment]
```
failed: brew linkage --test linux-pam

        Missing libraries:
  libdb-6.2.so
```

**Comment by maxim-belkin** [comment]
we need to rebuild linux-pam in `linuxbrew/extra` to switch to libdb-18.1.so from berkeley-db

**Comment by sjackman** [comment]
```
brew audit gpm
  * The URL http://www.nico.schottelius.org/software/gpm/ should use HTTPS rather than HTTP
```

**Comment by maxim-belkin** [comment]
wow :)

**Comment by maxim-belkin** [comment]
ah! I think SHA256 change is due to `full_index=1`

**Comment by maxim-belkin** [comment]
```
==> brew linkage --test jdk@7
Missing libraries:
  libGL.so.1
  libX11.so.6
  libXext.so.6
  libXi.so.6
  libXrender.so.1
  libXtst.so.6
  libXxf86vm.so.1
  libasound.so.2
  libatk-1.0.so.0
  libavcodec.so.52
  libavcodec.so.53
  libavformat.so.52
  libavformat.so.53
  libcairo.so.2
  libfontconfig.so.1
  libfreetype.so.6
  libgdk-x11-2.0.so.0
  libgdk_pixbuf-2.0.so.0
  libgio-2.0.so.0
  libglib-2.0.so.0
  libgmodule-2.0.so.0
  libgobject-2.0.so.0
  libgstreamer-lite.so
  libgthread-2.0.so.0
  libgtk-x11-2.0.so.0
  libjava.so
  libjvm.so
  libmawt.so
  libpango-1.0.so.0
  libpangocairo-1.0.so.0
  libpangoft2-1.0.so.0
  libxml2.so.2
  libxslt.so.1
Unwanted system libraries:
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
  libthread_db.so.1
```

**Comment by maxim-belkin** [comment]
```
$ ldd jre/lib/amd64/libprism-es2.so 
	linux-vdso.so.1 =>  (0x00007ffed677b000)
	libX11.so.6 => not found
	libXxf86vm.so.1 => not found
	libGL.so.1 => not found
	libc.so.6 => /lib/x86_64-linux-gnu/libc.so.6 (0x00007f64ab82c000)
	/lib64/ld-linux-x86-64.so.2 (0x00007f64abe00000)
```

**Comment by sjackman** [comment]
We ignore the `jdk` linkage errors. No problems occur under typical usage.

**Comment by sjackman** [comment]
```
brew fetch libuuid
Formula reports different SHA256: 46af3275291091009ad7f1b899de3d0cea0252737550e7919d17237997db5644
Downloaded to: /home/linuxbrew/.cache/Homebrew/libuuid--1.0.3.tar.gz
SHA256: 251a85b3bac687974f360d3796048c20ded3bf0bd69e0d1cfd1db23d013f89ed
```

**Comment by sjackman** [comment]
SourceForge again!

**Comment by sjackman** [comment]
Looks like SourceForge is back online.

File: Formula/libtirpc.rb (hunk 4 above)
Comment (maxim-belkin) [reply Formula/libtirpc.rb]:
```
checking gssapi/gssapi.h usability... no
checking gssapi/gssapi.h presence... no
checking for gssapi/gssapi.h... no
configure: error: gssapi.h not found. Use --disable-gssapi, or install GSS-API.
```
So, we can make it optional but then we have to add `--disable-gssapi`. Shall we do that?

**Comment by maxim-belkin** [comment]
@sjackman, this is good to look at, I think. It has to be pulled with the `--bottle` flag because of updated version of `libtirpc`. I can factor it out into a separate PR, if desired.

File: Formula/elfutils.rb [hunk 5]
```diff
@@ -15,8 +15,9 @@ class Elfutils < Formula
   depends_on "bzip2" unless OS.mac?
   depends_on "zlib" unless OS.mac?
   depends_on "valgrind" => [:build, :optional]
+  depends_on "m4" => :build
```
Comment (sjackman) [inline Formula/elfutils.rb]:
```ruby
unless OS.mac?
  depends_on "m4" => :build
  depends_on "bzip2"
  depends_on "zlib"
```

File: Formula/elfutils.rb [hunk 6]
```diff
@@ -32,6 +33,28 @@ def install
...
+      disasm-x86
+      readelf-self
+      strip-reloc
+      strip-strmerge
+    ]
+    skip_tests.each do |test|
+      file = "tests/run-#{test}.sh"
+      rm_f file
+      Pathname(file).write("exit 77", :perm => 0755)
+    end
```
Comment (sjackman) [inline Formula/elfutils.rb]: Add a blank line after this block please.

File: Formula/libfuse.rb [hunk 7]
```diff
@@ -27,14 +27,18 @@ def install
...
       #include <fuse.h>
-      #include <stdlib.h>
-      int main(int argc, char** argv)
-      {
-        struct fuse_operations ops = {NULL, };
+      static struct fuse_operations ops = {
+        .init = NULL,
+      };
+      int main(int argc, char** argv) {
+        umask(0);
```
Comment (sjackman) [inline Formula/libfuse.rb]: Please add a comment about why this line is needed.

Comment (sjackman) [reply Formula/libfuse.rb]: Something like `# Fix the error: foo: cannot frobnicate`

File: Formula/libgpm.rb [hunk 8]
```diff
@@ -1,17 +1,17 @@
...
-  desc "general purpose mouse"
-  homepage "http://www.nico.schottelius.org/software/gpm/"
-  url "http://www.nico.schottelius.org/software/gpm/archives/gpm-1.20.7.tar.bz2"
+  desc "General-purpose mouse"
+  homepage "https://www.nico.schottelius.org/software/gpm/"
+  url "https://www.nico.schottelius.org/software/gpm/archives/gpm-1.20.7.tar.bz2"
   sha256 "f011b7dc7afb824e0a017b89b7300514e772853ece7fc4ee640310889411a48d"
   # tag "linuxbrew"
 
+  head "https://github.com/telmich/gpm.git"
```
Comment (sjackman) [inline Formula/libgpm.rb]: Put this line immediately after `sha256`, no blank line needed.

File: Formula/libfuse.rb (hunk 7 above)
Comment (maxim-belkin) [reply Formula/libfuse.rb]: this like is not necessary in this example. I just looked up the examples in libfuse (https://github.com/libfuse/libfuse/blob/master/example/passthrough.c) and added it here. I'd like this test to be a bit more useful and actually execute the code but I'm not sure we can "execute" anything related to `libfuse` that without `sudo`

**Comment by sjackman** [comment]
```
failed: brew audit jdk@7 --online
failed: brew bottle --merge --write --no-commit ./libprelude--3.1.0_2.x86_64_linux.bottle.json --keep-old
failed: brew install --only-dependencies xmount
failed: brew install xmount
failed: brew linkage --test jdk@7
failed: brew style libtirpc
```
https://circleci.com/gh/Linuxbrew/homebrew-core/19596

`xmount` is resolved by PR https://github.com/Linuxbrew/homebrew-core/pull/9069
Ignoring issues with `jdk@7`

```
failed: brew style libtirpc
C: 20:  7: Style/UnneededInterpolation: Prefer to_s over string interpolation.
```

File: Formula/libtirpc.rb [hunk 9]
```diff
@@ -17,7 +17,7 @@ def install
       "--disable-dependency-tracking",
       "--disable-silent-rules",
       "--prefix=#{prefix}",
-      "#{"--disable-gssapi" if build.without? "krb5"}"
+      "--disable-gssapi" if build.without? "krb5"
```
Comment (sjackman) [inline Formula/libtirpc.rb]: Configure will not be run at all `if build.without? "krb5"`. You need parentheses.
```ruby
*("--disable-gssapi" if build.without? "krb5")
```
instead though I'd suggest the usual
```ruby
args << "--disable-gssapi" if build.without? "krb5"
```

File: Formula/libtirpc.rb [hunk 10]
```diff
@@ -10,14 +10,31 @@ class Libtirpc < Formula
...
+  depends_on "krb5" => :optional
 
   def install
     system "./configure",
-      "--disable-debug",
       "--disable-dependency-tracking",
       "--disable-silent-rules",
-      "--prefix=#{prefix}"
+      "--prefix=#{prefix}",
+      "--disable-gssapi" if build.without? "krb5"
```
Comment (sjackman) [inline Formula/libtirpc.rb]: Please remove this chunk unless it addresses a specific build failure.

Comment (sjackman) [reply Formula/libtirpc.rb]: Let's stick to style issues though in this PR.

**Comment by sjackman** [comment]
For future PRs, please use one PR per formula. This PR is unwieldily large.

File: Formula/libfuse.rb (hunk 7 above)
Comment (sjackman) [reply Formula/libfuse.rb]: Remove this line then. I'm happy if the test compiles. Running would be ideal, but like you say, may need sudo.

File: Formula/libtirpc.rb (hunk 9 above)
Comment (maxim-belkin) [reply Formula/libtirpc.rb]: there is no `args` variable

Comment (sjackman) [reply Formula/libtirpc.rb]: Please remove this chunk

File: Formula/libtirpc.rb (hunk 10 above)
Comment (maxim-belkin) [reply Formula/libtirpc.rb]: it fails to build without `krb5` if we don't specify `disable-gssapi`

File: Formula/libtirpc.rb (hunk 9 above)
Comment (maxim-belkin) [reply Formula/libtirpc.rb]: The formula fails to build without this line
```
checking gssapi/gssapi.h usability... no
checking gssapi/gssapi.h presence... no
checking for gssapi/gssapi.h... no
configure: error: gssapi.h not found. Use --disable-gssapi, or install GSS-API.
```

File: Formula/libtirpc.rb (hunk 10 above)
Comment (sjackman) [reply Formula/libtirpc.rb]: Ah, got it. In that case
```ruby
args = ["--prefix=#{prefix}", "--disable-debug", "--disable-dependency-tracking", "--disable-silent-rules"]
args << "--disable-gssapi" if build.without? "krb5"
system "./configure", *args
```
Please open a separate PR to address build failures though. Let's keep this PR to style.

**Comment by sjackman** [comment]
Great! Thanks!

**Pull request closed without merging**